app/assets/javascripts/pageflow/editor/models/file_uploader.js

```diff
@@ -33,7 +33,7 @@ pageflow.FileUploader = pageflow.Object.extend({
                                                   fileTypes: this.fileTypes}));
     }
 
-    this.entry.getFileCollection(fileType).add(file);
+    this.entry.getFileCollection(fileType).create(file);
```

Comment:
The promise returned by this method should only resolve after `create` invoked its `success` callback. Otherwise there is a chance of trying to access `s3_direct_upload_config` in `UploaderView` before it is available.